libcontainer/specconv/spec_linux.go

```diff
@@ -407,6 +407,19 @@ func createLibcontainerMount(cwd string, m specs.Mount) (*configs.Mount, error)
 			mnt.Source = filepath.Join(cwd, m.Source)
 		}
 	}
+
+	// None of the mount arguments can contain a null byte. Normally such
+	// strings would either cause some other failure or would just be truncated
+	// when we hit the null byte, but because we serialise these strings as
+	// netlink messages (which don't have special null-byte handling) we need
+	// to block this as early as possible.
+	if strings.Contains(mnt.Destination, "\x00") ||
+		strings.Contains(mnt.Source, "\x00") ||
+		strings.Contains(mnt.Device, "\x00") ||
+		strings.Contains(mnt.Data, "\x00") {
```

Comment:
It might be overkill to check *all* of these. We only strictly need to check `Source` but it seems reasonable to me to check at least `Destination` and `Device` as well.

Reply:
I was thinking if `Data` could contain a `NULL` byte. In mount(2) it is defined as `void *data` (rather than `char *data`), and it's said to be interpreted on a per-filesystem basis, so there's a possibility that it can be something with null bytes in the middle.

Practically, though, we always treat `Data` as a string with comma-separated options, and thus it makes sense to check it for `\0` as well.

Reply:
Yeah I was thinking of that too. Might be better to only check the definitely-string fields.